### ISSUE TITLE
Disks are Detected but Tests are not run.

### ISSUE THREAD

**Issue by mreed8855** [canonical/checkbox#255]
### Bug Description

In some Dell servers, drives are detected but the disk tests are skipped.  In these cases the test plan has not been modified.  

https://certification.canonical.com/hardware/202209-30647/submission/290366/
https://certification.canonical.com/hardware/202209-30651/submission/290331/

### To Reproduce

1.  sudo add-apt-repository ppa:hardware-certification/public
2.  sudo apt-get update
3.  sudo apt-get install checkbox-ng plainbox-provider-resource-generic plainbox-provider-certification-client plainbox-provider-checkbox
4.  Certify-22.04 or test-storage

### Environment

Jammy 22.04.1
Dell servers 

### Relevant log output

```shell
- Checkbox session(s) (located in `/var/tmp/checkbox-ng/sessions/`, you usually want to select the most recent one)

- logs from the impacted components (e.g. `lsblk` if this is related to an issue when testing a disk...); a safe option is to install and run `sosreport` to gather as much log as possible.
```

**Comment by bladernr** [comment]
This seems to be the udev for the affected disk:
    {
        "bus": "pci",
        "driver": "nvme",
        "path": "/devices/pci0000:00/0000:00:14.0/0000:05:00.0",
        "product_id": 8769,
        "subproduct_id": 8529,
        "subvendor_id": 4136,
        "vendor": "Marvell Technology Group Ltd.",
        "vendor_id": 6987,
        "vendor_slug": "Marvell_Technology_Group_Ltd."
    },
    {
        "bus": "nvme",
        "category": "OTHER",
        "driver": "nvme",
        "name": "nvme0",
        "path": "/devices/pci0000:00/0000:00:14.0/0000:05:00.0/nvme/nvme0",
        "product": "nvme0",
        "product_id": 6317,
        "product_slug": "nvme0",
        "subproduct_id": 29296,
        "subvendor_id": 32902,
        "vendor": "Marvell Technology Group Ltd.",
        "vendor_id": 32902,
        "vendor_slug": "Marvell_Technology_Group_Ltd."
    },
    {
        "bus": "nvme",
        "category": "DISK",
        "driver": "nvme",
        "path": "/devices/pci0000:00/0000:00:14.0/0000:05:00.0/nvme",
        "product_id": 8769,
        "subproduct_id": 8529,
        "subvendor_id": 4136,
        "vendor": "Marvell Technology Group Ltd.",
        "vendor_id": 6987,
        "vendor_slug": "Marvell_Technology_Group_Ltd."
    },

**Comment by mreed8855** [comment]
This appears to be an issue with the BOSS card.  Dell removed the BOSS card and re-ran the cert on the NVME drive and it passed.

 https://certification.canonical.com/hardware/202209-30647/submission/291825/

**Comment by mreed8855** [comment]
I cannot completely confirm that the BOSS card is the issue. In this certification on an mx840C the drive (sda) which is connected to the BOSS card passes.  https://certification.canonical.com/certificates/2211-13948/  It appears to be issues with specific NVME drives.   Dell Ent NVMe AGN SED RI U.2 3.84TB Dell Ent NVMe AGN SED MU U.2 12.8TB. These  drives with the "AGN" in there descriptions appear to be failing.

**Comment by mreed8855** [comment]
This does appear to be an issue with the Boss-N1 card.  However, the disk stress test does pass on these drives when run manually.  I think this is an issue with the script that I am trying to debug.  Here is another example of the issue. 
 https://certification.canonical.com/certificates/2302-14163/

**Comment by mreed8855** [comment]
This issue is with a specific nvme drive 182NW-Dell Ent NVMe v2 AGN RI U.2 15.36TB

**Comment by mreed8855** [comment]
I am seeing this issue on an HPE system where the nvme drive is detected but none of the storage tests are run.  https://certification.canonical.com/certificates/2302-14222/

**Comment by mreed8855** [comment]
I seeing this issue on another Dell system
Dell Technologies PowerEdge MX760C
https://certification.canonical.com/certificates/2303-14269/

**Comment by mreed8855** [comment]
I  am seeing this issue on this system as well.
Dell Technologies PowerEdge MX760C
https://certification.canonical.com/certificates/2303-14281/

**Comment by mreed8855** [comment]
These are different nvme drives than before that are not getting recognized by checkbox.  The drives do pass the disk test when run manually.
Dell Ent NVMe FIPS PM1735a MU 12.8TB 11.64 TiB
Dell Ent NVMe FIPS PM1733a RI 15.36TB 13.97 TiB
https://certification.canonical.com/certificates/2303-14282/

**Comment by kissiel** [comment]
I found the cause

```
path: /devices/pci0000:00/0000:00:1c.4/0000:08:00.0/nvme/nvme2
name: nvme2
bus: nvme
category: OTHER
driver: nvme
product_id: 31292
vendor_id: 32902
subproduct_id: 34946
subvendor_id: 4163
product: nvme2
vendor: Samsung Electronics Co Ltd
product_slug: nvme2
vendor_slug: Samsung_Electronics_Co_Lt
```
`category: OTHER is the culprit`
for a "normal" nvme it's:
```
path: /devices/pci0000:00/0000:00:1c.4/0000:08:00.0/nvme
bus: nvme
category: DISK
driver: nvme
product_id: 43044
vendor_id: 5197
subproduct_id: 43009
subvendor_id: 5197
vendor: Samsung Electronics Co Ltd
vendor_slug: Samsung_Electronics_Co_Ltd
```
so my mode of operation was:
I looked for the name of the job that's missing, found the template job in the base provider
then looked at what resource is used as the base for the template
in our case it's the template-resource: device
the next line in PXU filters out everything but the records that have an entry category == 'DISK'
with the filter: template-filter: device.category == 'DISK'
so I ran the device job with:
checkbox-cli run com.canonical.certification::device and searched for the nvme2
and there it was the "OTHER" category
to fix that 
checkbox/checkbox-support/checkbox_support/parsers/udevadm.py has the logic that classifies the devices

**Comment by mreed8855** [comment]
The BOSS-N1 is listed in many of the Dell certs on this C3 query

https://certification.canonical.com/devices/111371/reports?page=1

**Comment by rodwsmith** [comment]
Frank Heimes has also run into this on an IBM Z-system with a Saumsung MZWLJ7T6HALA-00007 NVMe device. Here's the certificate request:

https://certification.canonical.com/certificates/2411-16260/

**Comment by mreed8855** [comment]
Here is another instance of this issue.
https://certification.canonical.com/hardware/202410-35916/submission/402161/

**Comment by Hook25** [comment]
We have decided to clean up our issue tracker to systematize our approach to backlog grooming and planning. If this issue is still relevant to you, please reopen it and tag one of the maintainers or reach out on ~Checkbox on Mattermost. Going forward all issues that have received no activity for more than a year will be marked as 'stale' and closed after one week
